.github/workflows/cap-dev.yml

```diff
@@ -0,0 +1,42 @@
+name: CAP dev integration
+
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+
+permissions:
+  contents: read
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout cds-dbs
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js 22
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22'
+          cache: 'npm'
+
+      # Inject the GH_TOKEN_TOOLS credential for all calls to github.tools.sap
+      - name: Configure Git credentials for SAP GitHub Enterprise
+        run: |
+          git config --global url."https://x-access-token:${{ secrets.GH_TOKEN_TOOLS }}@github.tools.sap/".insteadOf "https://github.tools.sap/"
```

Comment:
When running in a forked repo, GH won't allow access to the secret so that the whole job will fail.  To prevent this, add [this check](https://github.com/cap-js/docs/blob/c8bccafafb98586b6ee2cd6f8a7dbb30d1b80fcd/.github/workflows/PR-SAP.yml#L15) to the job.  This will skip it if running from a fork.